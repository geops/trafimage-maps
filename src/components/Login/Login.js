import React, { useMemo, useCallback } from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { ReactComponent as SBBUser } from '../../img/sbb/user_92_large.svg';

import './Login.scss';

const Login = ({ appBaseUrl }) => {
  const { t } = useTranslation();
  const language = useSelector((state) => state.app.language);
  const permissionInfos = useSelector((state) => state.app.permissionInfos);

  const login = useMemo(() => {
    if (permissionInfos && permissionInfos.user) {
      return permissionInfos.user;
    }
    return t('Anmelden');
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [permissionInfos, language]);

  const openLoginPage = useCallback(() => {
    window.location.href = `${appBaseUrl}/login?next=${encodeURIComponent(
      window.location.href,
    )}`;
  }, [appBaseUrl]);

  return (
    <div
      className="wkp-login"
      role="button"
      onClick={openLoginPage}
      onKeyPress={(evt) => evt.which === 13 && openLoginPage()}
      tabIndex={0}
      title={permissionInfos && permissionInfos.user}
    >
      <SBBUser focusable={false} className="wkp-login-icon" />
      <span className="wkp-login-text">{login}</span>
    </div>
  );
};

Login.propTypes = {
  appBaseUrl: PropTypes.string.isRequired,
};

export default React.memo(Login);
