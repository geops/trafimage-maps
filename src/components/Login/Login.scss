@import '../../globals.scss';

.wkp-login {
  margin-right: 6px;
  padding-top: 2px;
  color: $brand-grey;
  display: flex;
  align-items: center;

  .wkp-login-icon {
    float: left;
    margin-bottom: 2px;
    height: 28px;
    width: 28px;

    path {
      stroke: $brand-grey;
    }
  }

  &:hover {
    color: $brand-primary-hover;

    svg {
      path {
        stroke: $brand-primary-hover;
      }
    }
  }
}
