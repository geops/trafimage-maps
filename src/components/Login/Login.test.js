import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Login from '.';

describe('Login', () => {
  describe('matches snapshot', () => {
    test('displaying default text', () => {
      const store = global.mockStore({
        app: {},
      });
      const component = renderer.create(
        <Provider store={store}>
          <Login appBaseUrl="http://foo.de" />
        </Provider>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('displaying user name', () => {
      const store = global.mockStore({
        app: { permissionInfos: { user: 'bar' } },
      });
      const component = renderer.create(
        <Provider store={store}>
          <Login appBaseUrl="http://foo.de" />
        </Provider>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
