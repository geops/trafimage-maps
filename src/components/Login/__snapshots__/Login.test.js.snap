// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot displaying default text 1`] = `
<div
  className="wkp-login"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <svg
    className="wkp-login-icon"
    focusable={false}
  >
    user_92_large.svg
  </svg>
  <span
    className="wkp-login-text"
  >
    Anmelden
  </span>
</div>
`;

exports[`Login matches snapshot displaying user name 1`] = `
<div
  className="wkp-login"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <svg
    className="wkp-login-icon"
    focusable={false}
  >
    user_92_large.svg
  </svg>
  <span
    className="wkp-login-text"
  >
    bar
  </span>
</div>
`;
